js/pro/krakenfutures.js

```diff
@@ -0,0 +1,239 @@
+                'watchOHLCV': false, // for now
+                'watchOrderBook': false, // for now
+                'watchOrders': false, // for now
+                'watchTicker': true,
+                'watchTickers': false, // for now
+                'watchTrades': false, // for now
+                // 'watchHeartbeat': true,
+                // 'watchStatus': true,
+            },
+            'urls': {
+                'api': {
+                    'ws': {
+                        'public': 'wss://futures.kraken.com/ws/v1',
+                    },
+                },
+            },
+            // 'versions': {
+            //     'ws': '0.2.0',
+            // },
+            'options': {
+                'tradesLimit': 1000,
+                'OHLCVLimit': 1000,
+                'ordersLimit': 1000,
+                'symbolsByOrderId': {},
+                'checksum': true,
+            },
+            'exceptions': {
+                'ws': {
+                    'exact': {
+                        'Event(s) not found': BadRequest,
+                    },
+                    'broad': {
+                        'Currency pair not in ISO 4217-A3 format': BadSymbol,
+                    },
+                },
+            },
+        });
+    }
+
+    handleTicker (client, message) {
+        //
+        //   {
+        //     "time": 1612270825253,
+        //     "feed": "ticker",
+        //     "product_id": "PI_XBTUSD",
+        //     "bid": 34832.5,
+        //     "ask": 34847.5,
+        //     "bid_size": 42864,
+        //     "ask_size": 2300,
+        //     "volume": 262306237,
+        //     "dtm": 0,
+        //     "leverage": "50x",
+        //     "index": 34803.45,
+        //     "premium": 0.1,
+        //     "last": 34852,
+        //     "change": 2.995109121267192,
+        //     "funding_rate": 3.891007752e-9,
+        //     "funding_rate_prediction": 4.2233756e-9,
+        //     "suspended": false,
+        //     "tag": "perpetual",
+        //     "pair": "XBT:USD",
+        //     "openInterest": 107706940,
+        //     "markPrice": 34844.25,
+        //     "maturityTime": 0,
+        //     "relative_funding_rate": 0.000135046879166667,
+        //     "relative_funding_rate_prediction": 0.000146960125,
+        //     "next_funding_rate_time": 1612281600000,
+        //     "volumeQuote": 262306237
+        //   }
+        //
+        const product_id = this.safeStringLower (message, 'product_id');
+        const name = 'ticker';
+        const market = this.market (product_id);
+        const symbol = market['symbol'];
+        const messageHash = name + ':' + symbol;
+        const timestamp = this.milliseconds ();
+        const result = {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': undefined,
+            'low': undefined,
+            'bid': this.safeFloat (message, 'bid'),
```

Comment:
use of `safeFloat` is deprecated you should use either `safeNumber` or `safeString`, since this is going to be the input of `safeTicker` we should use `safeString`